rivermigrate/migration/004_pending_and_more.up.sql

```diff
@@ -12,7 +12,7 @@ UPDATE river_job SET metadata = '{}' WHERE metadata IS NULL;
 ALTER TABLE river_job ALTER COLUMN metadata SET NOT NULL;
 
 -- The 'pending' job state will be used for upcoming functionality:
-ALTER TYPE river_job_state ADD VALUE 'pending' AFTER 'discarded';
+ALTER TYPE river_job_state ADD VALUE IF NOT EXISTS 'pending' AFTER 'discarded';
```

Comment:
Yeah, I don't see any other viable options.

The migration in its current form isn't right because it makes the rollback unusable given it can't be used to roll forward again.

A possibility is to to an enum swap by creating a whole new enum without `pending`, then swap it into place on `state`, and remove the old enum. I'm sure that'd need a full table scan to check the type though.

So yep, looks like this is the path.